src/core/ext/filters/client_channel/resolver/dns/c_ares/grpc_ares_wrapper.cc

```diff
@@ -487,6 +488,7 @@ static void grpc_ares_notify_on_event_locked(grpc_ares_ev_driver* ev_driver) {
 void grpc_ares_ev_driver_start_locked(grpc_ares_ev_driver* ev_driver) {
   grpc_ares_notify_on_event_locked(ev_driver);
   // Initialize overall DNS resolution timeout alarm
+  grpc_core::ExecCtx::Get()->InvalidateNow();
```

Comment:
This is not a repeated timer, so there can't be a loop here.

Reply:
The overall resolution timeout isn't repeated, but the backup poller alarm initiated just after is. I've removed the timer invalidation to be just above the spot where we initialize the backup polling loop

Reply:
Sorry on second though, I ended up going back and removing this, since even though this `grpc_timer_init` initiates the backup poller, the backup poller uses it's own callback and it's own `grpc_timer_init`, so this `grpc_timer_init` indeed isn't repeated.